use populateCache for optimistic UI in combination with remote mutation

uetrozi opened the pull request.

emyjamalian reviewed (changes requested):
We are using page routing and you have deleted the route `/api/tasks/[id]` but used the route `/api/tasks/${taskId}/toggleCompleted` in Tasklist for fetching that's why now delete function and marked as done don't really work cause the api calls are failing. 

BTW what was the issue with mutate? I think it was working well. or was there specific problems with it?

uetrozi commented:
I added the missing files again and changed some of the flickering.

So when I started with the inline editing, I noticed flickering in the checkboxes when you toggled them. There was also the problem (in my first attempt) that the done and upcoming pages were not working properly. Thats why I had a look at the SWR documentation ended up implementing the section about optimistic UI, because before, at least for me, the page reloaded on every edit.
